metaverse_components/wear.js

```diff
@@ -188,26 +195,68 @@ export default (app, component) => {
 
   const _copyBoneAttachment = spec => {
     const {boneAttachment = 'hips', position, quaternion, scale} = spec;
-    const boneName = Avatar.modelBoneRenames[boneAttachment];
-    const bone = player.avatar.foundModelBones[boneName];
-    if (bone) {
-      bone.matrixWorld
-        .decompose(app.position, app.quaternion, app.scale);
-      if (Array.isArray(position)) {
-        app.position.add(localVector.fromArray(position).applyQuaternion(app.quaternion));
-      }
-      if (Array.isArray(quaternion)) {
-        app.quaternion.multiply(localQuaternion.fromArray(quaternion));
-      }
-      if (Array.isArray(scale)) {
-        app.scale.multiply(localVector.fromArray(scale));
+    const boneAttachments = Array.isArray(boneAttachment) ? boneAttachment : [boneAttachment];
+
+    // lerp app's transform to average position/quaternion/scale of boneAttachments.
+    let count = 0;
+    boneAttachments.forEach((boneAttachment, i) => {
+      const boneName = Avatar.modelBoneRenames[boneAttachment];
+      const bone = player.avatar.foundModelBones[boneName];
+      if (bone) {
+        if (count === 0) {
+          bone.matrixWorld

```

Comment:
I think this could be very slow to decompose each bone. Is that really needed?

Reply:
It's derived from existing code https://github.com/webaverse/app/blob/295fb394e236aa40d8719ee28ec20fdc62990f72/metaverse_components/wear.js#L195 , and I think it is needed, because of will use them to do calculations later on https://github.com/webaverse/app/blob/295fb394e236aa40d8719ee28ec20fdc62990f72/metaverse_components/wear.js#L196-L204 .
And for now, need to lerp them through their lerp/slerp functions due to matrix has not lerp function.

But it's generally not a big performance issue, because of for new plants repo, will just loop two bones, not all the bones of avatar https://github.com/webaverse/plants/pull/1/files .
And for all old wearable repos, only has one single bone to `decompose`.

Deleted two other unneeded `decompose`s though https://github.com/webaverse/app/pull/3010/commits/2c96fbc466304768350e9ed88217b3d8f7fbcfa1 .